### PR TITLE
Get all categories

### PR DESCRIPTION
**Pull request by flamesoft**
PT story: https://www.pivotaltracker.com/story/show/125524401
- Set up test for the api endpoint
- Set up controller for the api endpoint

### CODE REVIEW

File: spec/requests/api/v1/category_data_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe 'Category Data', type: :request do
+
+  describe 'GET /api/v1/category_data' do
```
Comment (diraulo) [inline spec/requests/api/v1/category_data_spec.rb]: why `/api/v1/category_data` and not `/api/v1/categories`?

Comment (flamesoft) [inline spec/requests/api/v1/category_data_spec.rb]: We have one file in /api/v1/ called restaurant_data.rb, so we keep the same naming.

Comment (diraulo) [inline spec/requests/api/v1/category_data_spec.rb]: I'm talking more about the URL itself than the actual file (although there should be some consistency there too). There are some general rules when it comes to generating RESTful API URLs. 

In this particular case, you're returning  restaurant categories and as such:

`/api/v1/caterogies` or `/api/v1/restaurant_categories` makes more sense than `/api/v1/category_data`

**Comment by diraulo** [comment]
@flamesoft @callea2 I can merge this as it is but you should re-consider the way you name URLs to your resources. Here's a good article that sums up couple of best practices when working with APIs http://blog.mwaysolutions.com/2014/06/05/10-best-practices-for-better-restful-api/

**Comment by flamesoft** [comment]
Thanks @diraulo for the naming tips. I have renamed the category_data to categories.

**Comment by tochman** [comment]
I would suggest that you do an overview of the other endpoint as well.

**Comment by flamesoft** [comment]
It works with test locally. The semaphoreci has some problem with installation. I have to restart it manually sometimes to make it work.

**Comment by tochman** [comment]
It goes green after the route update. Great stuff.

**Pull request merged**